include/spla-cpp/SplaLibrary.hpp

```diff
@@ -43,7 +45,102 @@ namespace spla {
      */
     class SPLA_API Library {
     public:
+        /**
+         * @class Config
+         *
+         * Represents system configuration used in all computations.
+         * @p Config initializes OpenCL context, which includes all
+         * specified devices.
+         */
+        class SPLA_API Config {
+        public:
+            /**
+             * Type of OpenCL device.
+             */
+            enum DeviceType {
+                GPU,
+                CPU,
+                Accelerator,
+            };
+
+            /**
+             * Amount of used OpenCL devices.
+             */
+            enum DeviceAmount {
+                One,
+                All,
+            };
+
+            /**
+             * Creates @p Config with all specified devices.
+             *
+             * @param devices Vector of device names which will be used in context.
+             *
+             * @note No checks about devices are performed, so
+             * first make sure that all @p devices can be used in
+             * the same context.
+             */
+            explicit Config(const std::vector<std::string> &devices);
+
+            /**
+             * Creates @p Config with only one default device.
+             */
+            explicit Config();
+
```

Comment:
Config will have another entries, not only device/platfrom selecton related stuff.
So it is good to have only single constructor and a bunch of `Setters`, where we
can set values. 

```cpp
Config& SetPlatformHint(const std::string& platformHintName);
Config& SetDeviceHint(DeviceAmount amount);
// etc.
Library::Config cfg;
cfg.SetPlatformHint(/** smth */)
   .SetDeviceHint(/** smth */);
```